homeassistant/components/shelly/climate.py

```diff
@@ -440,9 +442,7 @@ def __init__(
         elif entry is not None:
             self._unique_id = entry.unique_id
         self._attr_device_info = get_entity_block_device_info(coordinator, sensor_block)
-        self._attr_name = get_block_entity_name(
-            self.coordinator.device, sensor_block, None
-        )
+        self._attr_name = get_block_channel_name(self.coordinator.device, sensor_block)
```

Comment:
This is the main device entity: 
```suggestion
        self._attr_name = None
```

Ref: https://developers.home-assistant.io/docs/core/entity/#has_entity_name-true-mandatory-for-new-integrations